R/utility_spliceHelper_functions.R

```diff
@@ -1,598 +1,572 @@
+      #queryLastJunction <- ranges(selectEndExonsFromGrangesList(splice, exonNumber = 1))
+
+      startMatch <- poverlaps(unlist(queryStart), unlist(subjectStart))
+      endMatch <- poverlaps(unlist(queryEnd), unlist(subjectEnd))
+
+      #calculate distance between first and last exon matches:
+      # distances corresponds to length of unique sequences in all matching exons
+      # distance = width(element in query) + width(all matching elements in subject) - 2x width(intersection(query, subject)
+      subjectList <- unlist(subjectFull)
+      queryStartList <- rep(unlist(queryStart),elementNROWS(subjectFull))
+      myId <- rep(1:length(queryStart),elementNROWS(subjectFull))
+      byExonIntersect=pintersect(queryStartList,subjectList, resolve.empty='start.x')
+      startDist <- as.integer(tapply(width(subjectList)*(width(byExonIntersect)>0)-2*width(byExonIntersect),myId,sum) + width(unlist(queryStart)))
+      uniqueStartLengthQuery <-   sum(width(GenomicRanges::setdiff(queryStart, subjectFull)))
+      uniqueStartLengthSubject <-  startDist - uniqueStartLengthQuery
+
+
+
+      queryEndList <- rep(unlist(queryEnd),elementNROWS(subjectFull))
+      myId <- rep(1:length(queryEnd),elementNROWS(subjectFull))
+      byExonIntersect=pintersect(queryEndList,subjectList, resolve.empty='start.x')
+      endDist <- as.integer(tapply(width(subjectList)*(width(byExonIntersect)>0)-2*width(byExonIntersect),myId,sum) + width(unlist(queryEnd)))
+      uniqueEndLengthQuery <-   sum(width(GenomicRanges::setdiff(queryEnd, subjectFull)))
+      uniqueEndLengthSubject <-  endDist - uniqueEndLengthQuery
+    } else {
+      startMatch <- NULL
+      uniqueStartLengthQuery <- NULL
+      uniqueStartLengthSubject <- NULL
+      endMatch <- NULL
+      uniqueEndLengthQuery <- NULL
+      uniqueEndLengthSubject <- NULL
+    }
 
-  exon5Prime <- tapply(queryExonStart, querySplice.subjectFullQuerySplice.IdMap, sum)
-  exon3Prime <- tapply(queryExonEnd, querySplice.subjectFullQuerySplice.IdMap, sum)
+    mcols(olap) <- DataFrame( mcols(olap), startMatch, uniqueStartLengthQuery,uniqueStartLengthSubject,endMatch, uniqueEndLengthQuery,uniqueEndLengthSubject)
+  }
 
-  subjectStartExonStartExtension <- start(subjectStart.querySplice.Rng)<end(qSpRngList) & end(subjectStart.querySplice.Rng)>end(qSpRngList) & start(qSpRngList)< start(subjectStart.querySplice.Rng)
-  subjectStartExonStartExtension <- tapply(subjectStartExonStartExtension, subjectStartEnd.querySplice.IdMap, sum)
+  olap
+}
 
-  subjectStartExonEndExtension <- end(subjectStart.querySplice.Rng)>start(qSpRngList) & start(subjectStart.querySplice.Rng)<start(qSpRngList) & end(qSpRngList)> end(subjectStart.querySplice.Rng)
-  subjectStartExonEndExtension <- tapply(subjectStartExonEndExtension, subjectStartEnd.querySplice.IdMap, sum)
+#' annotate splice overlap by distance
+#' @noRd
+annotateSpliceOverlapsByDist <-function(query, subject) {
```

Comment:
choose a more meaningful name? what about compareTranscripts()?
Can you add the documentation for this function, including the example with the example data?

Reply:
it might be nice to illustrate how to use this in the bambu workflow. Do you want to create an issue for that?

Reply:
I have added an [issue](https://github.com/GoekeLab/bambu/issues/178) to address this.